src/main/java/movie/storage/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package movie.storage.dao.impl;
+
+import java.util.List;
+import movie.storage.dao.MovieDao;
+import movie.storage.exception.IncorrectDataHandlingException;
+import movie.storage.lib.Dao;
+import movie.storage.model.Movie;
+import movie.storage.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = HibernateUtil.getSessionFactory().openSession();
+        try {
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```

Comment:
You don't need to set id manually, Hibernate does it automatically

Reply:
agree

Reply:
Done

Reply:
has it?